### ISSUE TITLE
Error recording analysis telemetry

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#2867]
### Description

The telemetry XML file can be corrupted, which then causes SLVS to throw when trying to record subsequent analyses.

**Root cause:  **
The invalid XML file below from my machine, but I haven't been able to reproduce the steps that cause it.

Prior to the exception I had opened an C++ project in a VS instance that does not have the C++ workload installed, so I assumed the bug was related to #2853. However, I haven't managed to repro it subsequently.

**Impact: **
The exceptions are caught and handled internally i.e. no immediately visible user impact.
However, the bug prevents subsequent telemetry about language analysis from being recorded correctly. It is not clear whether analysis data is lost only until the next "reset" of the telemetry (i.e. when the file is reset after the data is uploaded), or permanently.

#### Example of invalid XML

[telemetry - bad.xml.txt](https://github.com/SonarSource/sonarlint-visualstudio/files/8069402/telemetry.-.bad.xml.txt)

```
  <Analyses>
    <Analysis xsi:nil="true" />
    <Analysis>
      <Language>cpp</Language>
    </Analysis>
  </Analyses>
```

#### Subsequent exception
 
![image](https://user-images.githubusercontent.com/10757559/154071278-e0d7cad8-5bee-4b80-a0e6-81865dd66267.png)

### Affected versions
Initially observed in VS2019. The file is in the per-user roaming profile, so it will affect all of instances of VS for that user on any machine they use in the same domain.
